ui/src/components/Pin.tsx

```diff
@@ -58,8 +75,10 @@ export function Pin({
       state === PinState.Selected ? new L.Point(48, 57) : new L.Point(39, 48),
   });
 
-  const leftPaddingPoint = L.point(500, 75);
-  const rightPaddingPoint = L.point(100, 400);
+  const leftPaddingPoint =
+    window.screen.height <= parseInt(device.mobile) ? null : L.point(500, 75);
+  const rightPaddingPoint =
```

Comment:
Hmm we might have to put this in a https://reactjs.org/docs/hooks-reference.html#uselayouteffect hook. (also will address on window resize) So something like:

```ts
// outside of component
const LEFT_PADDING_POINT = L.point(500, 75);
const RIGHT_PADDING_POINT = L.point(100, 400);

type PaddingState =
  | { left: L.Point; right: L.Point }
  | { left: null; right: null };

// inside component
const [paddingPoints, setPaddingPoints] = React.useState<PaddingState>({
  left: LEFT_PADDING_POINT,
  right: RIGHT_PADDING_POINT,
});

React.useLayoutEffect(() => {
  function updateSize() {
    if (window.screen.height <= parseInt(device.mobile)) {
      setPaddingPoints({left: null, right: null});
    } else {
      setPaddingPoints({left: LEFT_PADDING_POINT, right: RIGHT_PADDING_POINT});
    }
  }
  window.addEventListener("resize", updateSize);
  updateSize();
  return () => window.removeEventListener("resize", updateSize);
});

// return
<StyledPopup
  autoPanPaddingTopLeft={paddingPoints.left}
  autoPanPaddingBottomRight={paddingPoints.right}
>
```

I wrote this whole thing in GitHub, so there is probably syntax errors and other stuff, but hopefully this gives you a good direction

Reply:
actually we might have to compute this outside of the Pin so that we dont add an event listener for each pin